app/lib/features/events/providers/event_providers.dart

```diff
@@ -95,7 +99,9 @@ final allPastEventListProvider = FutureProvider.autoDispose
     .family<List<ffi.CalendarEvent>, String?>((ref, spaceId) async {
   final allEventList = await ref.watch(allEventListProvider(spaceId).future);
   List<ffi.CalendarEvent> allPastEventList = allEventList
-      .where((event) => getEventType(event) == EventFilters.past)
+      .where(
+        (event) => ref.watch(eventTypeProvider(event)) == EventFilters.past,
```

Comment:
I wonder if overload not occurred when a `ref.watch()` is placed in other `ref.watch()`.